src/DurableTask.Core/TaskOrchestrationDispatcher.cs

```diff
@@ -563,7 +566,7 @@ async Task<OrchestrationExecutionCursor> ExecuteOrchestrationAsync(Orchestration
             // Get the TaskOrchestration implementation. If it's not found, it either means that the developer never
             // registered it (which is an error, and we'll throw for this further down) or it could be that some custom
             // middleware (e.g. out-of-process execution middleware) is intended to implement the orchestration logic.
-            TaskOrchestration? taskOrchestration = this.objectManager.GetObject(runtimeState.Name, runtimeState.Version!);
+            TaskOrchestration? taskOrchestration = this.objectManager.GetObject(runtimeState.Name, runtimeState.Version!, this.DataConverter);
```

Comment:
We can avoid adding to object manager contract by setting the property ourselves here:

```suggestion
            TaskOrchestration? taskOrchestration = this.objectManager.GetObject(runtimeState.Name, runtimeState.Version!);
            if (taskOrchestration is IDataConverterConfigurable configurable)
            {
                configurable.DataConverter = this.dataConverter;
            }
```